cd/ack-chart/update-chart.sh

```diff
@@ -264,7 +267,41 @@ _commit_chart_changes() {
     popd >/dev/null
 }
 
+_poll_for_upgraded_chart() {
+    local triggering_chart_dir
+    triggering_chart_dir="$WORKSPACE_DIR/$REPO_NAME"
+
+    local newest_chart_version
+    newest_chart_version="$(yq '.version' "$triggering_chart_dir/helm/Chart.yaml")"
+
+    echo "Fetching ECR public bearer token"
+
+    local tag_list_authentication_token
+    tag_list_authentication_token=$(curl -ss -k https://public.ecr.aws/token/ | jq -r '.token')
+
+    local chart_name
+    chart_name="aws-controllers-k8s/${REPO_NAME//-controller/-chart}"
+
+    echo "Waiting until $chart_name@$newest_chart_version becomes available on ECR public"
+    # Use SECONDS as timeout counter
+    until curl -ss -k -H "Authorization: Bearer $tag_list_authentication_token" \
+        "https://public.ecr.aws/v2/$chart_name/tags/list" \
+        | jq --exit-status --arg CHART_VERSION "$newest_chart_version" \
+        'any(.tags[]; . == $CHART_VERSION)' &> /dev/null; do
```

Comment:
Until relatively recently ECR Public didn't support listing tags, so this is a welcome addition. Nice bash-fu, too :)

Reply:
Thank you Jesse Butler. He led that effort :)